ImageAnalysis/image_analysis/analyzers/UC_GenericLightSpecCam.py

```diff
@@ -130,6 +141,54 @@ def analyze_image(self, input_image: Array2D, auxiliary_data: Optional[dict] = N
                                                          input_parameters=self.build_input_parameter_dictionary())
         return return_dictionary
 
+    def perform_spectral_correction(self, wavelength, spectrum, image,
+                                    eff_low_bound=0.025):
+
+        # load detection efficiency
+        spectral_correction = self.construct_spectral_correction(wavelength)
```

Comment:
Consider calling construct_spectral_correction() from __init__ instead of from perform_spectral_correction(). That way, if the same analyzer is used on many (hundreds) of images, the spectral correction isn't constructed every time.

Reply:
Agreed.  While in the live python analysis generates a new analyzer for every image because the python session only persists for that single image from Labview, being able to reuse the same analyzer in post-analysis scripts has been quite convenient from my experience.